controllers/profile_controller.go

```diff
@@ -65,12 +67,14 @@ func (r *ProfileSubscriptionReconciler) Reconcile(ctx context.Context, req ctrl.
 	pSub := v1alpha1.ProfileSubscription{}
 	err := r.Client.Get(ctx, client.ObjectKey{Name: req.Name, Namespace: req.Namespace}, &pSub)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Info("resource has been deleted")
+			return ctrl.Result{}, nil
+		}
 		logger.Error(err, "failed to get resource")
 		return ctrl.Result{}, err
 	}
 
-	// TODO delete if deletion timestamp != nil
-
 	// TODO replace with mutating webhook
```

Comment:
AFAIK since we don't use finalizers the deletion timestamp doesn't get set, the resource gets immediately removed. So deletions are handled by checking if the `get` fails because the resource does not exist.